json_serialization/reader.nim

```diff
@@ -527,23 +566,27 @@ proc readValue*[T](r: var JsonReader, value: var T)
     when expectedFields > 0:
       let fields = T.fieldReadersTable(ReaderType)
       var expectedFieldPos = 0
-      while r.lexer.tok == tkString:
+      while r.lexer.lazyTok == tkString:
```

Comment:
How about removing the `lexer.strVal` field altogether by replacing its usages by a scanning procedure that can be called directly in the `readValue` functions?

Similarly, instead of having the `lexer.accept` procedure, we could have a more sophisticated `scanNumber` helper that is used by the `readValue` functions directly. It can either have multiple output variables indicating the type of number that was scanned or it could be a template where the user passes in different blocks of code that will be responsible for handling each number type (this is a more complicated approach that can prove to be more performant).

Reply:
These approaches have the added benefit that they would allow the user to process JSON data without allocating any memory for strings.

Reply:
ok, that is a good point.

I did lazy string scanning but removed it. My point was that double processing a string is not a problem. But I see that a Unix kernel dump :) should probably not be held in memory.

Will look into that.